test/k8sT/Nightly.go

```diff
@@ -292,21 +292,26 @@ var _ = Describe("NightlyEpsMeasurement", func() {
 			testNcConnectivity(360)
 		}
 
-		It("Test TCP Keepalive with L7 Policy", func() {
-			kubectl.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
-			manifest := helpers.ManifestGet(netcatDsManifest)
-			kubectl.Apply(manifest).ExpectSuccess("Cannot apply netcat ds")
-			defer kubectl.Delete(manifest)
-			testConnectivity()
-		})
+		Context("Test TCP Keepalive", func() {
 
-		It("Test TCP Keepalive without L7 Policy", func() {
-			manifest := helpers.ManifestGet(netcatDsManifest)
-			kubectl.Apply(manifest).ExpectSuccess("Cannot apply netcat ds")
-			defer kubectl.Delete(manifest)
-			kubectl.Exec(fmt.Sprintf(
-				"%s delete --all cnp -n %s", helpers.KubectlCmd, helpers.DefaultNamespace))
-			testConnectivity()
+			BeforeEach(func() {
+				kubectl.Apply(netcatDsManifest).ExpectSuccess("Cannot apply netcat ds")
+			})
+
+			AfterEach(func() {
+				_ = kubectl.Delete(netcatDsManifest)
+			})
+
+			It("With L7 Policy", func() {
+				kubectl.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
+				testConnectivity()
+			})
+
+			It("Without L7 Policy", func() {
+				kubectl.Exec(fmt.Sprintf(
```

Comment:
I do not really liked this one. Could we use`ciliumPolicyAction` that waits until all endpoints are regenerated?

Reply:
Makes sense. The only concern I have here is that I might be changing the functionality of the test, and I have no real way of validating the nightly tests locally (not enough resources), and nightly tests are not stable on master. How do you recommend that I validate that I'm not breaking the test ?